docs/reference_guide.md

```diff
@@ -842,7 +843,26 @@ Attaching 1 probe...
 ^C
 ```
 
-## 13. Looping Constructs
+## 13. Array casts
+
+It is possible to cast between integer arrays and integers. Both the source and
+the destination type must have the same size. The main purpose of this is to
+allow casts from/to byte arrays.
+
+Example:
+
+```
+# bpftrace -e 'BEGIN { $a = (int8[8])12345; printf("%x %x\n", $a[0], $a[1]); printf("%d\n", (uint64)$a); }'
+Attaching 1 probe...
+39 30
+12345
+^C
```

Comment:
Throw in the pton example too? I find that to be a good motivating example

Reply:
Good idea, will add it.